### ISSUE TITLE
label selector problem when running xgboost and sklearn iris prepack examples together

### ISSUE THREAD

**Issue by ryandawsonuk** [SeldonIO/seldon-core#777]
If I run either one on its own then it works. But if I run both together I get

```
{"level":"error","ts":1565364741.1764412,"logger":"kubebuilder.controller","msg":"Reconciler error","controller":"seldondeployment-controller","request":"seldon/xgboost","error":"Deployment.apps \"iris-default\" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{\"seldon-app\":\"xgboost-iris-default\", \"seldon-app-classifier\":\"iris-default-classifier-seldonio-xgboostserver-rest-0-1\", \"seldon-deployment-id\":\"iris-xgboost\"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable","stacktrace":"github.com/seldonio/seldon-operator/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/github.com/seldonio/seldon-operator/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/seldonio/seldon-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/seldonio/seldon-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:217\ngithub.com/seldonio/seldon-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\t/go/src/github.com/seldonio/seldon-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\ngithub.com/seldonio/seldon-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/seldonio/seldon-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\ngithub.com/seldonio/seldon-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/seldonio/seldon-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\ngithub.com/seldonio/seldon-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/seldonio/seldon-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
```

I think this means there are non-unique names in the labels. Presumably related to them both using iris-default and possibly also related to https://github.com/SeldonIO/seldon-operator/pull/47
